src/tsdb/src/tsdbRead.c

```diff
@@ -986,7 +986,9 @@ static SMemRow getSMemRowInTableMem(STableCheckInfo* pCheckInfo, int32_t order,
       return rmem;
     } else {
       pCheckInfo->chosen = CHECKINFO_CHOSEN_BOTH;
-      *extraRow = rimem;
+      if (extraRow) {
```

Comment:
It is potential when there is the same key in mutable and immutable memory tables when update is set to be 2 for partial updating but extraRow is NULL.